pkg/pulumiyaml/run_invoke_test.go

```diff
@@ -143,6 +143,72 @@ runtime: yaml
 	requireNoErrors(t, tmpl, diags)
 }
 
+func TestInvokeVariableSugar(t *testing.T) {
+	t.Parallel()
+
+	const text = `
+name: test-yaml
+runtime: yaml
+variables:
+  foo:
+    Fn::test:invoke:type:
+      quux: tuo
+resources:
+  res-a:
+    type: test:resource:type
+    properties:
+      foo: ${foo.retval}
+`
+
+	tmpl := yamlTemplate(t, strings.TrimSpace(text))
+	diags := testInvokeDiags(t, tmpl, func(r *runner) {})
+	requireNoErrors(t, tmpl, diags)
+}
+
+func TestInvokeOutputVariableSugar(t *testing.T) {
+	t.Parallel()
+
+	const text = `
+name: test-yaml
+runtime: yaml
+variables:
+  foo:
+    Fn::test:invoke:type:
+      quux: ${res-a.out}
+resources:
+  res-a:
+    type: test:resource:type
+    properties:
+      foo: oof
+  res-b:
+    type: test:resource:type
+    properties:
+      foo: ${foo.retval}
+`
+
+	tmpl := yamlTemplate(t, strings.TrimSpace(text))
+	diags := testInvokeDiags(t, tmpl, func(r *runner) {})
+	requireNoErrors(t, tmpl, diags)
+}
+
+func TestInvokeNoInputsSugar(t *testing.T) {
+	t.Parallel()
+
+	const text = `
+variables:
+  config:
+    Fn::test:invoke:empty:
```

Comment:
I'm surprised that this is valid. 😀 I would have expected that you needed something like `Fn::test:invoke:empty: {}`

Reply:
Don't think we have to modify to error if `{}` isn't provided, but it looks nicer so we should probably use it in future examples